### PR TITLE
Generation and Tree

### LINKED ISSUE #2: [FEATURE] Create a basic tree structure and display it using ASCII
This feature summarizes the basic use of the module which gets all the information through api commands, and outputs the information to the console. This is helpful for both setting up the initial module, testing that it is working as intended, and adding some basic functionality to the module

### PR DESCRIPTION
**Pull request by porrige51122**
This pr contains all the OU generation code which is a custom struct to hold all the information needed for this tool. It also includes all the functions to populate this struct with actual account data.

It also has a CLI display tool added to it so the user can see the output of the tool in a visually pleasing way.
This closes #2.

### CODE REVIEW

File: generation/ou_struct.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// style of a tree.
+type OU struct {
+	Id       string
+	Name     string
+	Children []*OU
+	Accounts []types.Account
+}
+
+// addChildren adds the given OUs to the OU's children slice.
+func (o *OU) addChildren(children []*OU) {
```
Comment (KaeDig) [inline generation/ou_struct.go:22]: Oh wow this is now giving me serious Java vibes... Getters and setters? What a blast!

Comment (KaeDig) [reply generation/ou_struct.go:22]: (Realised I highlighted the wrong part of the code there, but I was referring to all the other getters in that file)

File: display/cli/display.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+package cli
+
+import (
+	"github.com/CentricaDevOps/aws-organizations-visualiser/generation"
+)
+
+// Display is a function that takes in the tree structure and displays it in a
+// visually appealing way in the CLI.
+func Display(tree *generation.OU) {
+	// TODO: use flags to determine whether to display detailed output or not
```
[the remark this answers is not in the document]
Comment (porrige51122) [reply display/cli/display.go:15]: Completely intentional. Code isn't finished yet, but this pr completes a vertical slice of the product

File: display/cli/tree.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/CentricaDevOps/aws-organizations-visualiser/generation"
+)
+
+// tree is a struct that holds the information needed to display the tree in the
+// CLI. It is used with the setupTreeRecursive and printTreeRecursive functions.
+type tree struct {
+	referencedNode *generation.OU
```
[the remark this answers is not in the document]
Comment (porrige51122) [reply display/cli/tree.go:13]: There will be a json export but that will be a different package. This struct is private so there aren't any intentions of external reusability of the struct

File: generation/ou_struct.go (hunk 1 above)
Comment (porrige51122) [reply generation/ou_struct.go:22]: Private setters public getters. Just makes creating other parts of the program a bit more clear.

**Pull request merged**